igvc_gazebo/nodes/control/control.cpp

```diff
@@ -171,6 +136,48 @@ int main(int argc, char **argv)
     rate.sleep();
     prev = now;
   }
+}
+
+void Control::speedCallback(const igvc_msgs::velocity_pair::ConstPtr &msg)
+{
+  if (msg->left_velocity == msg->left_velocity)
+  {
+    speed_set_point_left = msg->left_velocity;
+  }
+  if (msg->right_velocity == msg->right_velocity)
+  {
+    speed_set_point_right = msg->right_velocity;
+  }
+}
+
+void Control::jointStateCallback(const sensor_msgs::JointStateConstPtr &msg)
+{
+  auto iter = std::find(msg->name.begin(), msg->name.end(), std::string{ "left_axle" });
+
+  if (iter != msg->name.end())
+  {
+    auto index = std::distance(msg->name.begin(), iter);
+
+    speed_measured_left = (msg->velocity[index]) * (wheel_radius);
+  }
+
+  iter = std::find(msg->name.begin(), msg->name.end(), std::string{ "right_axle" });
+
+  if (iter != msg->name.end())
+  {
+    auto index = std::distance(msg->name.begin(), iter);
+
+    speed_measured_right = (msg->velocity[index]) * (wheel_radius);
+  }
+
+  // ROS_INFO_STREAM("right speed: " << speed_measured_right << " left speed: " << speed_measured_left);
+}
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "jessii_controller");
+
+  Control c;
```

Comment:
Currently `c` is uninitialized. Make sure to add a constructor otherwise nothing will happen.